src/rsa.rs

```diff
@@ -333,14 +360,21 @@ impl RSAKeyPair {
     /// platforms, it is done less perfectly. To help mitigate the current
     /// imperfections, and for defense-in-depth, base blinding is always done.
     /// Exponent blinding is not done, but it may be done in the future.
-    pub fn sign(&self, padding_alg: &'static RSAPadding,
+    pub fn sign(&self, padding_alg: &'static pad::PaddingAlgorithm,
                 rng: &rand::SecureRandom, msg: &[u8], signature: &mut [u8])
                 -> Result<(), ()> {
         if signature.len() != self.public_modulus_len() {
             return Err(());
         }
 
-        try!(padding_alg.pad(msg, signature));
+        try!(padding_alg.encode(msg, signature, rng));
+
+        let mod_bits = self.public_modulus_len_bits() - 1;
```

Comment:
A workaround to handle PSS padding which is aligned to the number of bits in the modulus.

An alternative would be to add an input variable to encode which states the length in bits the output should be.